src/mc_tasks/lipm_stabilizer/StabilizerTask.cpp

```diff
@@ -724,15 +835,71 @@ sva::ForceVecd StabilizerTask::computeDesiredWrench()
   dcmVelError_ = dcmDerivator_.eval();
 
   Eigen::Vector3d desiredCoMAccel = comddTarget_;
-  desiredCoMAccel += omega_ * (c_.dcmPropGain * dcmError_ + comdError);
+  desiredCoMAccel += omega_ * (c_.dcmPropGain * dcmError_ + c_.comdErrorGain * comdError);
   desiredCoMAccel += omega_ * c_.dcmIntegralGain * dcmAverageError_;
   desiredCoMAccel += omega_ * c_.dcmDerivGain * dcmVelError_;
-  auto desiredForce = mass_ * (desiredCoMAccel + constants::gravity);
+  desiredCoMAccel -= omega_ * omega_ * c_.zmpdGain * zmpdTarget_;
+
+  // Calculate CoM offset from measured wrench
+  extWrenchesMeasured_.clear();
+  for(const auto & extWrenchTarget : extWrenchesTarget_)
+  {
+    extWrenchesMeasured_.emplace_back(
+        extWrenchTarget.first,
+        sva::ForceVecd(extWrenchGain_.vector().cwiseProduct(robot().surfaceWrench(extWrenchTarget.first).vector())));
```

Comment:
This implies that all bodies provided in `setExternalWrenches` must have an attached force sensor (either directly or indirectly).

I suggest we check if `robot().surfaceHasForceSensor(extWrenchTarget.first) || robot().surfaceHasIndirectForceSensor(extWrenchTarget.first)` and act as if the measured wrench is equal to the target wrench otherwise.

This way external wrenches can be provided without a force sensor.

Reply:
Actually we only need to check for `robot().surfaceHasIndirectForceSensor(extWrenchTarget.first)`. Its implementation already takes care of this optimization.

Reply:
Fixed in https://github.com/jrl-umi3218/mc_rtc/pull/95/commits/a1eb8a66aab082c0aa3f080b1b47d7a91e815e61